packages/@aws-cdk/aws-scheduler-alpha/README.md

```diff
@@ -138,6 +138,21 @@ new Schedule(this, 'Schedule', {
 });
 ```
 
+### Configuring a start and end time of the Schedule
+
+If you choose a recurring schedule, you can set the start and end time of the Schedule by specifying the startDate and endDate.
+These values must follow the ISO 8601 format (`yyyy-MM-ddTHH:mm:ss.SSSZ`).
+
+```ts
+declare const target: targets.LambdaInvoke;
+
+new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.rate(cdk.Duration.hours(12)),
+    target: target,
+    startDate: '2023-01-01T00:00:00.000Z',
+    endDate: '2023-02-01T00:00:00.000Z',
```

Comment:
I think we standardized on using the `Date` type a while back. Similar to what we are doing [here](https://github.com/aws/aws-cdk/blob/15c7bb2908cff1ab8bebacacc09f26bbab6ec391/packages/aws-cdk-lib/aws-applicationautoscaling/lib/scalable-target.ts#L196). We can then convert to whatever format scheduler expects.

How do you feel about that? I think that's the best course of action here as it allows for more types of inputs, but feel free to push back.

Reply:
Thanks for the comment!

Using the Date type to allow for a variety of user inputs is certainly appealing.
However, I have some reservations about it, primarily due to potential issues with timezones.
When we initialize a Date object with a string that lacks timezone specification(`new Date("2022-08-04T15:19:46.125")`), the system's timezone is used by default.
For instance, if a local machine is set to Asia/Tokyo and the CI environment to UTC, the same time would be offset by 9 hours between these environments.
Since we are only dealing with ISO 8601 format and exclusively in UTC for this properties, using a string might be a safer option to avoid these timezone pitfalls.

By typing these values as string representation in the ISO 8601 format, we could mitigate these inconsistencies and ensure more predictable behavior across different environments.

This is just my personal opinion and I would love to get feedback from the CDK team.

Reply:
CFN accepts `startDate` only in UTC time. I think when we do `new Date().toISOString()` the Date is automatically converted to UTC time, specifically ISO 8601, which is what we want.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString

Reply:
Yes, `toISOString` method is used to convert to ISO 8601 format (with UTC timezone), but in the previous comment I was referring to the timezone at the initialization of the Date object.

However, I would like to change these properties to use Date type, since this problem does not occur if the user passes a string with timezone at initialization, and this is not a CDK problem but a JavaScript Date class specification problem.
Sorry for the confusion.